File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace GridViewExpandedGroupRowsStateMvc
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial(ByVal customAction As String) As ActionResult
			ViewData("actionToPerform") = customAction
			Return PartialView()
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;

namespace GridViewExpandedGroupRowsStateMvc {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult GridViewPartial(string customAction) {
            ViewData["actionToPerform"] = customAction;
            return PartialView();
        }
    }
}